facebook/delphiFacebook/R/contingency_tables.R

```diff
@@ -0,0 +1,1068 @@
+    stop("no input files to read in")
+  }
+  
+  params$start_time <- date_range[[1]]
+  params$end_time <- date_range[[2]]
+  
+  params$start_date <- as.Date(date_range[[1]])
+  params$end_date <- as.Date(date_range[[2]])
+  
+  return(params)
+}
+
+
+#' Get relevant input data file names from `input_dir`.
+#'
+#' @param date_range    List of two dates specifying start and end of desired
+#' date range 
+#' @param params    Params object produced by read_params
+#'
+#' @return Character vector of filenames
+#' 
+#' @export
+get_filenames_in_range <- function(date_range, params) {
+  start_date <- as.Date(date_range[[1]]) - params$archive_days
+  end_date <- as.Date(date_range[[2]])
+  date_pattern = "^[0-9]{4}-[0-9]{2}-[0-9]{2}.*[.]csv$"
+  youtube_pattern = ".*YouTube[.]csv$"
+  
+  filenames <- list.files(path=params$input_dir)
+  filenames <- filenames[grepl(date_pattern, filenames) & !grepl(youtube_pattern, filenames)]
+  
+  file_end_dates <- as.Date(substr(filenames, 1, 10))
+  file_start_dates <- file_end_dates
+  
+  # Only keep files with data that falls at least somewhat between the desired
+  # start and end range dates.
+  filenames <- filenames[
+    !(( file_start_dates < start_date & file_end_dates < start_date ) | 
+        ( file_start_dates > end_date & file_end_dates > end_date ))]
+  
+  return(filenames)
+}
+
+
+#' Run the contingency table production pipeline
+#'
+#' See the README.md file in the source directory for more information about how to run
+#' this function.
+#'
+#' @param params    Params object produced by read_params
+#' @param aggregations Data frame with columns `name`, `var_weight`, `metric`,
+#'   `group_by`, `compute_fn`, `post_fn`. Each row represents one aggregate
+#'   to report. `name` is the aggregate's base column name; `var_weight` is the 
+#'   column to use for its weights; `metric` is the column of `df` containing the
+#'   response value. `group_by` is a list of variables used to perform the 
+#'   aggregations over. `compute_fn` is the function that computes
+#'   the aggregate response given many rows of data. `post_fn` is applied to the
+#'   aggregate data after megacounty aggregation, and can perform any final
+#'   calculations necessary.
+#'
+#' @return none
+#' 
+#' @importFrom parallel detectCores
+#' 
+#' @export
+run_contingency_tables <- function(params, aggregations)
+{
+  params <- update_params(params)
+  
+  cw_list <- produce_crosswalk_list(params$static_dir)
+  archive <- load_archive(params)
+  msg_df("archive data loaded", archive$input_data)
+  
+  input_data <- load_responses_all(params)
+  input_data <- filter_responses(input_data, params)
+  msg_df("response input data", input_data)
+  
+  input_data <- merge_responses(input_data, archive)
+  data_agg <- create_data_for_aggregatation(input_data)
+  
+  data_agg <- filter_data_for_aggregatation(data_agg, params, lead_days = 12)
+  data_agg <- join_weights(data_agg, params, weights = "step1")
```

Comment:
Note to self: I'm checking with Facebook on whether it makes more sense to use the full weights here instead of step1, since we're using responses from all over the survey to make the contingency tables

Reply:
Facebook agrees that switching to the full weights makes sense